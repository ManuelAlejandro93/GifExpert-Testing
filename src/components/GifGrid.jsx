import PropTypes from 'prop-types';

import GifItem from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

const GifGrid = ({ title = defaultProps.title }) => {
	const { gifsData, isLoading } = useFetchGifs(title);

	return (
		<>
			<h2>{title}</h2>
			{isLoading && <h1>Trayendo la info de: {title}</h1>}
			{!isLoading && (
				<div className='card-grid '>
					{gifsData.map(gif => (
						<GifItem
							key={gif.id}
							{...gif} /* manda todas las props por cada imagen */
						/>
					))}
				</div>
			)}
		</>
	);
};

export default GifGrid;

GifGrid.propTypes = {
	title: PropTypes.string.isRequired,
};

const defaultProps = {
	title: 'Russia',
};
