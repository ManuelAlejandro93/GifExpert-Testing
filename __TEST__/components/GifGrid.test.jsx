import { render, screen } from '@testing-library/react';
import GifGrid from '../../src/components/GifGrid';

describe('<GifGrid/>', () => {
	// test('Generar un snapshot del componente', () => {
	// 	const { container } = render(<GifGrid title='Naruto' />);
	// 	expect(container).toMatchSnapshot();
	// });
	test('Debe traer la info de carga de datos', () => {
		const { container } = render(<GifGrid title='Naruto' />);
		const infoCargando = screen.getAllByText(/Trayendo la info de/i);

		expect(infoCargando).toBeTruthy();
	});
	test('Debe traer la info del titulo Naruto', () => {
		const { container } = render(<GifGrid title='Naruto' />);
		const infoNaruto = screen.getAllByText(/Naruto/i);

		expect(infoNaruto).toBeTruthy();
	});
});
