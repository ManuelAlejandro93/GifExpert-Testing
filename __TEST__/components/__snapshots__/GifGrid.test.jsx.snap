// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGrid/> Generar un snapshot del componente 1`] = `
<div>
  <h2>
    Naruto
  </h2>
  <h1>
    Trayendo la info de: 
    Naruto
  </h1>
</div>
`;
